common/Reducers.jsm

```diff
@@ -500,10 +500,22 @@ function DiscoveryStream(prevState = INITIAL_STATE.DiscoveryStream, action) {
         ...prevState,
         feeds: {
           ...prevState.feeds,
-          data: action.data || prevState.feeds.data,
           loaded: true,
         },
       };
+    case at.DISCOVERY_STREAM_FEED_UPDATE:
+      const newData = {};
+      newData[action.data.url] = action.data.feed;
```

Comment:
is it possible to have action.data empty here, if yes we should have a safe check

Reply:
Not currently possible, the place we call it is explicitly creating the data object when it calls dispatch, it's also not called if url doesn't connect with a feed. Feed itself could be undefined if the server is misconfigured and doesn't return anything originally, but I imagine that's actually fine here because we don't directly reference something on feed, we just pass it along. If the server starts doing that, we probably got much bigger problems :D